tools/Azure.Mcp.Tools.EventHubs/src/Commands/Namespace/NamespaceGetCommand.cs

```diff
@@ -0,0 +1,140 @@
+    {
+        base.RegisterOptions(command);
+        command.Options.Add(OptionDefinitions.Common.ResourceGroup.AsOptional());
+        command.Options.Add(EventHubsOptionDefinitions.NamespaceName.AsOptional());
+    }
+
+    protected override NamespaceGetOptions BindOptions(ParseResult parseResult)
+    {
+        var options = base.BindOptions(parseResult);
+        options.ResourceGroup ??= parseResult.GetValueOrDefault<string>(OptionDefinitions.Common.ResourceGroup.Name);
+        options.NamespaceName = parseResult.GetValueOrDefault<string>(EventHubsOptionDefinitions.NamespaceName.Name);
+        return options;
+    }
+
+    public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+    {
+        if (!Validate(parseResult.CommandResult, context.Response).IsValid)
+        {
+            return context.Response;
+        }
+
+        var options = BindOptions(parseResult);
+
+        try
+        {
+            var eventHubsService = context.GetService<IEventHubsService>();
+
+            // Determine if this is a single namespace request or list request
+            bool isSingleNamespaceRequest = !string.IsNullOrEmpty(options.NamespaceName) && !string.IsNullOrEmpty(options.ResourceGroup);
+
+            if (isSingleNamespaceRequest)
+            {
+                // Get single namespace with detailed information
+                var namespaceDetails = await eventHubsService.GetNamespaceAsync(
+                    options.NamespaceName!,
+                    options.ResourceGroup!,
+                    options.Subscription!,
+                    options.Tenant,
+                    options.RetryPolicy);
+
+                context.Response.Results = namespaceDetails != null
+                    ? ResponseResult.Create(new(namespaceDetails), EventHubsJsonContext.Default.NamespaceGetSingleCommandResult)
+                    : null;
+            }
+            else
+            {
+                var namespaces = await eventHubsService.GetNamespacesAsync(
+                    options.ResourceGroup!,
+                    options.Subscription!,
+                    options.Tenant,
+                    options.RetryPolicy);
+
+                context.Response.Results =  ResponseResult.Create(new(namespaces ?? []), EventHubsJsonContext.Default.NamespaceGetCommandResult);
+            }
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error getting Event Hubs namespaces");
+            HandleException(context, ex);
+        }
+
+        return context.Response;
+    }
+
+    protected override HttpStatusCode GetStatusCode(Exception ex) => ex switch
+    {
+        KeyNotFoundException => HttpStatusCode.NotFound,
+        RequestFailedException reqEx => (HttpStatusCode)reqEx.Status,
+        Identity.AuthenticationFailedException => HttpStatusCode.Unauthorized,
+        ArgumentException => HttpStatusCode.BadRequest,
```

Comment:
This check is already done in base.GetStatusCode()... unsure about the others but worth checking.

Reply:
If it's done in base I'll remove it